src/Microsoft.AspNetCore.Proxy/ProxyServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿
+using System;
+using Microsoft.AspNetCore.Proxy;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class ProxyServiceCollectionExtensions
+    {
+        public static IServiceCollection AddProxy(this IServiceCollection services)
+        {
+            return services.AddSingleton(new ProxyService(new SharedProxyOptions()));
```

Comment:
> Moving this to a central service and options prevents an app from proxying to more than one back-end that has different settings (e.g. configuring the HttpClientHandler differently). E.g. using different credentials, etc..
1. Credentials should be configured at HttpClient not at the handler.
2. But if you really need dedicated handler you can use AddScoped :). So no trade-off here at all